### ISSUE TITLE
socket path issue on MacOS Apple Silicon

### ISSUE THREAD

**Issue by theganyo** [containers/podman-desktop#1767]
### Bug description

After install, seeing this on the dashboard:

```
⚠️ Docker Socket Compatibility: Podman is not emulating the default Docker socket path: /var/run/docker.sock. Docker-specific tools may not work. See troubleshooting page on podman-desktop.io for more information.
```

Checking the referenced file:

```
$ ls -l /var/run/docker.sock
lrwxr-xr-x  1 root  daemon  66 Mar 16 08:46 /var/run/docker.sock -> /Users/theganyo/.local/share/containers/podman/machine/podman.sock
$ ls -l /Users/theganyo/.local/share/containers/podman/machine/podman.sock
lrwxr-xr-x  1 theganyo  primarygroup  89 Mar 15 17:05 /Users/theganyo/.local/share/containers/podman/machine/podman.sock -> /Users/theganyo/.local/share/containers/podman/machine/podman-machine-default/podman.sock
$ ls -l /Users/theganyo/.local/share/containers/podman/machine/podman-machine-default/podman.sock
srw-------@ 1 theganyo  primarygroup  0 Mar 16 08:46 /Users/theganyo/.local/share/containers/podman/machine/podman-machine-default/podman.sock
```

Following troubleshooting instructions to recreate doesn't fix the issue: (https://podman-desktop.io/docs/troubleshooting#warning-about-docker-compatibility-mode. )

I see this on start:

```
$ podman machine start
Starting machine "podman-machine-default"
Waiting for VM ...
Mounting volume... /Users:/Users
Error: invalid configuration: the specified mapping 100000:1000000 in "/etc/subuid" includes the user UID
Error: invalid configuration: the specified mapping 100000:1000000 in "/etc/subuid" includes the user UID
Mounting volume... /private:/private
Error: invalid configuration: the specified mapping 100000:1000000 in "/etc/subuid" includes the user UID
Error: invalid configuration: the specified mapping 100000:1000000 in "/etc/subuid" includes the user UID
Mounting volume... /var/folders:/var/folders
Error: invalid configuration: the specified mapping 100000:1000000 in "/etc/subuid" includes the user UID
Error: invalid configuration: the specified mapping 100000:1000000 in "/etc/subuid" includes the user UID
WARN[0039] API socket failed ping test

This machine is currently configured in rootless mode. If your containers
require root permissions (e.g. ports < 1024), or if you run into compatibility
issues with non-podman clients, you can switch using the following command:

	podman machine set --rootful

API forwarding listening on: /var/run/docker.sock
Docker API clients default to this address. You do not need to set DOCKER_HOST.

Machine "podman-machine-default" started successfully
```

I see this on attempting to run example:

```
$ podman run quay.io/podman/hello
Cannot connect to Podman. Please verify your connection to the Linux system using `podman system connection list`, or try `podman machine init` and `podman machine start` to manage a new Linux VM
Error: unable to connect to Podman socket: Get "http://d/v4.4.1/libpod/_ping": ssh: rejected: connect failed (open failed)

$ podman system connection list
Name                         URI                                                            Identity                                     Default
podman-machine-default       ssh://core@localhost:58443/run/user/434053/podman/podman.sock  /Users/theganyo/.ssh/podman-machine-default  true
podman-machine-default-root  ssh://root@localhost:58443/run/podman/podman.sock              /Users/theganyo/.ssh/podman-machine-default  false
```

### Operating system

MacOS 13.2.1 Apple silicon

### Version

0.12.0

**Comment by cdrage** [comment]
Hi @theganyo We've actually added an automated fix for this within: https://github.com/containers/podman-desktop/pull/1697

Are you able to try the latest (main) build and see if it works for you? It'll automatically run `podman-mac-helper` and diagnose the socket issues.

You may also need to update `podman` to the latest version as there *was* an issue with regards to socket compatibility before (unsure what version it was off hand that had the issue).

**Comment by theganyo** [comment]
Hey @cdrage. So I installed the latest build from your releases page, deleted the default machine, started podman desktop and told it to initialize and start. The end result was the same error.

I then installed the latest release of podman (4.4.2) and did the above again... this time I got a new error:

```
Error: Starting machine "podman-machine-default" Waiting for VM ... Error: qemu exited unexpectedly with exit code -1, stderr: dyld[50468]: Library not loaded: /usr/local/opt/capstone/lib/libcapstone.4.dylib Referenced from: <EB32AE6A-9195-392B-9581-BABFBD39C9F0> /opt/podman/qemu/bin/qemu-system-x86_64 Reason: tried: '/usr/local/opt/capstone/lib/libcapstone.4.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/opt/capstone/lib/libcapstone.4.dylib' (no such file), '/usr/local/opt/capstone/lib/libcapstone.4.dylib' (no such file), '/usr/lib/libcapstone.4.dylib' (no such file, not in dyld cache)
```

**Comment by benoitf** [comment]
@theganyo https://github.com/containers/podman/issues/17490

**Comment by theganyo** [comment]
Thanks, @benoitf. It sounds like either downgrading podman to 4.3.1 or waiting for 4.4.3 is the workaround for now.

**Comment by benoitf** [comment]
@theganyo if you have `brew` you can also install qemu from brew

**Comment by theganyo** [comment]
I wouldn't normally want to install brew just for that, but coincidentally I just had to install brew a couple days ago for another project, so I just tried installing qemu via brew. It didn't change the podman error. I think I'll try downgrading.

**Comment by benoitf** [comment]
@theganyo try to install podman and qemu from brew

**Comment by benoitf** [comment]
and remove `/opt/podman/`

**Comment by theganyo** [comment]
Thanks, but no dice. This seems like it may be a rabbit hole.

```
$ podman machine start
Starting machine "podman-machine-default"
Waiting for VM ...
Error: qemu exited unexpectedly with exit code 1, stderr: qemu-system-x86_64: invalid accelerator hvf
qemu-system-x86_64: falling back to tcg
qemu-system-x86_64: unable to find CPU model 'host'
```

**Comment by benoitf** [comment]
https://podman-desktop.io/docs/troubleshooting#podman-machine-on-apple-silicon

**Comment by theganyo** [comment]
FYI: It looks like downgrading is not a viable option, either. I removed podman and qemu from homebrew then downgraded to the podman 4.3.1 release, but ended up with the same issue that I started with here.

**Comment by theganyo** [comment]
FYI on 4.3.1, this is what I'm seeing at the CLI. But I guess none of this is directly podman-desktop and perhaps I should open a separate issue for podman directly...

```
Starting machine "podman-machine-default"
Waiting for VM ...
Mounting volume... /Users/theganyo:/Users/theganyo
Error: invalid configuration: the specified mapping 100000:1000000 in "/etc/subuid" includes the user UID
Error: invalid configuration: the specified mapping 100000:1000000 in "/etc/subuid" includes the user UID
WARN[0039] API socket failed ping test

This machine is currently configured in rootless mode. If your containers
require root permissions (e.g. ports < 1024), or if you run into compatibility
issues with non-podman clients, you can switch using the following command:

	podman machine set --rootful

API forwarding listening on: /var/run/docker.sock
Docker API clients default to this address. You do not need to set DOCKER_HOST.

Machine "podman-machine-default" started successfully
```

I can connect via ssh:

```
$ podman machine ssh
Connecting to vm podman-machine-default. To close connection, use `~.` or `exit`
Error: invalid configuration: the specified mapping 100000:1000000 in "/etc/subuid" includes the user UID
[core@localhost ~]$ exit
logout
```

But docker functions remain broken:

```
$ podman run quay.io/podman/hello
Cannot connect to Podman. Please verify your connection to the Linux system using `podman system connection list`, or try `podman machine init` and `podman machine start` to manage a new Linux VM
Error: unable to connect to Podman socket: Get "http://d/v4.3.1/libpod/_ping": ssh: rejected: connect failed (open failed)

**Comment by benoitf** [comment]
are you able to run `podman run quay.io/podman/hello` inside the VM when you do ssh

And what is the output of `podman system connections list` on your macOS host. Is the default conn cation correct ?

**Comment by theganyo** [comment]
```
$ podman machine ssh
Connecting to vm podman-machine-default. To close connection, use `~.` or `exit`
Last login: Wed Mar 22 11:10:43 2023 from 192.168.127.1
Error: invalid configuration: the specified mapping 100000:1000000 in "/etc/subuid" includes the user UID
[core@localhost ~]$ podman run quay.io/podman/hello
Error: invalid configuration: the specified mapping 100000:1000000 in "/etc/subuid" includes the user UID
```
```
$ podman system connection list
Name                         URI                                                            Identity                                     Default
podman-machine-default       ssh://core@localhost:55137/run/user/434053/podman/podman.sock  /Users/theganyo/.ssh/podman-machine-default  true
podman-machine-default-root  ssh://root@localhost:55137/run/podman/podman.sock              /Users/theganyo/.ssh/podman-machine-default  false
```

(BTW: I don't have a `/etc/subuid` file.)

**Comment by theganyo** [comment]
Oops, that was not quite accurate. While my laptop doesn't have a `/etc/subuid`, the podman machine does:

```
[core@localhost ~]$ cat /etc/subuid
core:100000:1000000
[core@localhost ~]$
```

**Comment by theganyo** [comment]
The 4.3.1 problem was that my user uid on my host machine within the range that the `/etc/subuid` declared on the podman machine. Changing this range fixed the issue. Note: I haven't tried upgrading to 4.4.2 as I expect I would to see the original qemu issue in that case.

reference: https://github.com/containers/podman/issues/17893

**Comment by theganyo** [comment]
Since the qemu issue is a [known issue](https://github.com/containers/podman/issues/17490) and should be fixed in [4.4.3](https://github.com/containers/podman-desktop/issues/1798), I'll close this.
